src/main/java/gregtech/common/covers/ender/CoverAbstractEnderLink.java

```diff
@@ -0,0 +1,411 @@
+                        .asWidget()
+                        .background(GTGuiTextures.SLOT)
+                        .size(18)
+                        .marginRight(2);
+    }
+
+    protected IWidget createPrivateButton() {
+        var isPrivate = new BooleanSyncValue(this::isPrivate, this::setPrivate);
+        isPrivate.updateCacheFromSource(true);
+
+        return new ToggleButton()
+                .tooltip(tooltip -> tooltip.setAutoUpdate(true))
+                .background(GTGuiTextures.PRIVATE_MODE_BUTTON[0])
+                .hoverBackground(GTGuiTextures.PRIVATE_MODE_BUTTON[0])
+                .selectedBackground(GTGuiTextures.PRIVATE_MODE_BUTTON[1])
+                .selectedHoverBackground(GTGuiTextures.PRIVATE_MODE_BUTTON[1])
+                .tooltipBuilder(tooltip -> tooltip.addLine(IKey.lang(this.isPrivate ?
+                        "cover.ender_fluid_link.private.tooltip.enabled" :
+                        "cover.ender_fluid_link.private.tooltip.disabled")))
+                .marginRight(2)
+                .value(isPrivate);
+    }
+
+    protected IWidget createIoRow() {
+        var ioEnabled = new BooleanSyncValue(this::isIoEnabled, this::setIoEnabled);
+
+        return new Row().marginBottom(2)
+                .coverChildrenHeight()
+                .child(new ToggleButton()
+                        .value(ioEnabled)
+                        .overlay(IKey.dynamic(() -> IKey.lang(this.ioEnabled ?
+                                "behaviour.soft_hammer.enabled" :
+                                "behaviour.soft_hammer.disabled").get())
+                                .color(Color.WHITE.darker(1)))
+                        .widthRel(0.6f)
+                        .left(0));
+    }
+
+    @Override
+    public boolean isWorkingEnabled() {
+        return workingEnabled;
+    }
+
+    @Override
+    public void setWorkingEnabled(boolean isActivationAllowed) {
+        this.workingEnabled = isActivationAllowed;
+    }
+
+    public boolean isIoEnabled() {
+        return ioEnabled;
+    }
+
+    protected void setIoEnabled(boolean ioEnabled) {
+        this.ioEnabled = ioEnabled;
+    }
+
+    private boolean isPrivate() {
+        return isPrivate;
+    }
+
+    private void setPrivate(boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        updateLink();
+        writeCustomData(UPDATE_PRIVATE, buffer -> buffer.writeBoolean(this.isPrivate));
+    }
+
+    @Override
+    public void writeInitialSyncData(PacketBuffer packetBuffer) {
+        packetBuffer.writeString(this.playerUUID == null ? "null" : this.playerUUID.toString());
+    }
+
+    @Override
+    public void readInitialSyncData(PacketBuffer packetBuffer) {
+        // does client even need uuid info? just in case
+        String uuidStr = packetBuffer.readString(36);
+        this.playerUUID = uuidStr.equals("null") ? null : UUID.fromString(uuidStr);
+        // client does not need the actual tank reference, the default one will do just fine
```

Comment:
hanging comment?

Reply:
idk, it was like that in `CoverEnderFluidLink`. could remove it